app/src/main/java/com/github/doomsdayrs/apps/shosetsu/backend/TarArchive.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.doomsdayrs.apps.shosetsu.backend
+
+import kotlinx.coroutines.yield
+import org.kamranzafar.jtar.*
+import java.io.File
+import java.io.InputStream
+import java.io.RandomAccessFile
+
+class TarArchive(file:String):File(file) {
+
+
+    fun read(name: String): InputStream {
+        val tar = TarInputStream(this.inputStream())
+        var buffer = ByteArray(0)
+        val entries =  generateSequence { tar.nextEntry }
+
+        entries.find {  it.name.contains(name,true) }?.run {
```

Comment:
instead of using run block

why not just do the following:

`val file = entries.find{...}`

and the use file to do the rest of the computation.

---

Same advice for every other place you're using run block.